parsers/ERP_PGCB.py

```diff
@@ -214,12 +215,24 @@ def fetch_production(
         # infer 'unknown'
         if row["total_generation"] is not None:
             # Total generation includes all sources, including imports so we need to subtract them to get the unknown source
-            unknown_source_mw = (
-                row["total_generation"]
-                - known_sources_sum_mw
-                - row["bd_import_bheramara"]
-                - row["bd_import_tripura"]
-            )
+            # Before this date Adani import in NoneType
+            if target_datetime is not None and target_datetime < datetime(2024, 8, 28):
+                unknown_source_mw = (
+                    row["total_generation"]
+                    - known_sources_sum_mw
+                    - row["bd_import_bheramara"]
+                    - row["bd_import_tripura"]
+                )
+
+            else:
+                unknown_source_mw = (
+                    row["total_generation"]
+                    - known_sources_sum_mw
+                    - row["bd_import_bheramara"]
+                    - row["bd_import_tripura"]
+                    - row["bd_import_adani"]
+                )
```

Comment:
Would something like this be simpler? That way, we don't need to write the entire "formula" twice.

```suggestion
              unknown_source_mw = (
                    row["total_generation"]
                    - known_sources_sum_mw
                    - row["bd_import_bheramara"]
                    - row["bd_import_tripura"]
                )

            # Adani import was added after this date
            if target_datetime is None or target_datetime > datetime(2024, 8, 28):
                unknown_source_mw -=  row["bd_import_adani"]
```